src/monocle_apptrace/exporters/azure/azure_exporter.py

```diff
@@ -0,0 +1,142 @@
+import os
+from opentelemetry.sdk.trace import ReadableSpan
+from opentelemetry.sdk.trace.export import SpanExporter, SpanExportResult
+from azure.storage.blob import BlobServiceClient, BlobClient, ContainerClient
+from azure.core.exceptions import ResourceNotFoundError, ClientAuthenticationError, ServiceRequestError
+import datetime
+from typing import Sequence
+import logging
+import time
+import random
+
+# Configuration
+DEFAULT_FILE_PREFIX = "monocle_trace_"
+DEFAULT_TIME_FORMAT = "%Y-%m-%d_%H.%M.%S"
+MAX_BATCH_SIZE = 500  # Max number of spans per batch
+MAX_QUEUE_SIZE = 10000  # Max number of spans in the queue
+EXPORT_INTERVAL = 1  # Maximum interval (in seconds) to wait before exporting spans
+BACKOFF_FACTOR = 2
+MAX_RETRIES = 10
+
+# Set up logging
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+class AzureBlobSpanExporter(SpanExporter):
+    def __init__(self, connection_string=None, container_name=None):
+        # Use default values if none are provided
+        if connection_string is None:
+            connection_string = os.getenv('AZURE_STORAGE_CONNECTION_STRING')
+            if connection_string is None:
```

Comment:
You can use 
if not connection_string:

if not container_name:


This checks for empty string as well